src/error-once.md

```diff
@@ -0,0 +1,112 @@
+# Handle Errors Once
+
+When a function receives an error from a caller,
+it can handle it in a variety of different ways
+depending on what it knows about the error.
+
+These include, but not are limited to:
+
+- if the function contract defines specific errors,
+  matching the error with `errors.Is` or `errors.As`
+  and handling the branches differently
+- if the error is recoverable,
+  logging the error and degrading gracefully
+- if the error represents a domain-specific failure condition,
+  returning a well-defined error
+- [wrapping the error](error-wrap.md) and returning it
+- returning the error as-is
```

Comment:
optional: since "wrapping the error" also talks about returning errors as-is so consider if these should be merged into one